src/content/reference/rsc/server-actions.md

```diff
@@ -103,32 +103,33 @@ export async function updateName(name) {
 }
 ```
 
-```js [[1, 3, "updateName"], [1, 13, "updateName"], [2, 11, "submitAction"],  [2, 23, "submitAction"]]
+```js [[1, 4, "updateName"], [1, 14, "updateName"], [2, 11, "submitAction"],  [2, 24, "submitAction"]]
 "use client";
 
-import {updateName} from './actions';
+import { useState, useTransition } from 'react';
+import { updateName } from './actions';
 
 function UpdateName() {
-  const [name, setName] = useState('');
   const [error, setError] = useState(null);
 
   const [isPending, startTransition] = useTransition();
 
-  const submitAction = async () => {
+  const submitAction = async (formData) => {
     startTransition(async () => {
-      const {error} = await updateName(name);
-      if (!error) {
-        setError(error);
+      const name = formData.get('name');
+      const result = await updateName(name);
+      if (result?.error) {
+        setError(result.error);
       } else {
-        setName('');
```

Comment:
When the form submission succeeds, React already automatically resets the form. No need to reset manually the input value. However, we need to reset the error.

Reply:
"When the form submission succeeds, React already automatically resets the form." sorry where's that claim coming from? In [the official docs](https://react.dev/reference/react-dom/components/form) there's an example doing `formRef.current.reset();`, I guess exactly because it is not reset by default? (thought seeing the other mistakes there, it might be?).

Reply:
Ah gotcha, thanks! I am just a passerby that was curious about the PR though, cannot review/approve/etc here, but looks good!